impeller/renderer/backend/vulkan/command_buffer_cache.h

```diff
@@ -0,0 +1,93 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <optional>
+
+#include "flutter/impeller/renderer/backend/vulkan/vk.h"
+
+namespace impeller {
+
+template <typename CommandBuffer>
```

Comment:
The only reason this is templated is to serve the mock. This is unfortunate as it affects our ability to OOL this if needed and makes the code harder to read. Barring another way to test it that doesn't use templates, I'd rather leave the test out and make this simpler (and OOLed). If we mess up our bindings, I'm sure higher level tests will catch this.

Reply:
Done.